core/src/smartcontracts/isi/asset.rs

```diff
@@ -451,8 +494,51 @@ pub mod query {
                 .evaluate(wsv, &Context::default())
                 .wrap_err("Failed to get account id")
                 .map_err(|e| Error::Evaluate(e.to_string()))?;
-            iroha_logger::trace!(%id);
-            wsv.account_assets(&id).map_err(Into::into)
+
+            let sort_by_key = match self.sort_by_key.as_ref() {
+                Some(name) => Some(
+                    name.evaluate(wsv, &Context::default())
+                        .wrap_err("Failed to get sorting key")
+                        .map_err(|e| Error::Evaluate(e.to_string()))?,
+                ),
+                None => None,
+            };
+
+            iroha_logger::trace!(%id, ?sort_by_key);
+
+            let mut vec = wsv.account_assets(&id).map_err(Error::from)?;
+
+            if let Some(sort_by_key) = sort_by_key {
+                vec.sort_by(|left, right| {
+                    let left = match left.value() {
+                        AssetValue::Store(metadata) => {
+                            metadata.get(&sort_by_key).unwrap_or(&Value::U128(0))
+                        }
+                        _ => &Value::U128(0),
+                    };
+
+                    let left = match left {
+                        Value::U128(value) => value,
+                        _ => &0,
+                    };
+
+                    let right = match right.value() {
+                        AssetValue::Store(metadata) => {
+                            metadata.get(&sort_by_key).unwrap_or(&Value::U128(0))
+                        }
+                        _ => &Value::U128(0),
+                    };
+
+                    let right = match right {
+                        Value::U128(value) => value,
+                        _ => &0,
+                    };
+
+                    left.cmp(right)
+                });
+            }
```

Comment:
comparison functions are the same, please extract them into a freestanding function

Reply:
Better yet, remove them into the query endpoing module, so they work independently of the query.